src/pd/pd.rs

```diff
@@ -378,7 +385,11 @@ impl<T: PdClient> Runner<T> {
                             PD_VALIDATE_PEER_COUNTER_VEC
                                 .with_label_values(&["peer stale"])
                                 .inc();
-                            send_destroy_peer_message(ch, local_region, peer, pd_region);
+                            if let Some(source) = merge_source {
+                                send_merge_fail(ch, source);
```

Comment:
How do we handle the TODO on line 407?

https://github.com/pingcap/tikv/pull/2502/files#diff-a2b393fbe5f436bb88e05e100af8c1b4R407

Reply:
It will be discussed later.